service/resource/deploymentv3/resource.go

```diff
@@ -89,7 +105,31 @@ func getDeploymentByName(list []*v1beta1.Deployment, name string) (*v1beta1.Depl
 }
 
 func isDeploymentModified(a, b *v1beta1.Deployment) bool {
-	return !reflect.DeepEqual(a.Spec.Template.Spec, b.Spec.Template.Spec)
+	aVersion, ok := a.GetAnnotations()[VersionBundleVersionAnnotation]
+	if !ok {
```

Comment:
why does not have OR condition instead of multiple if?

Reply:
Based on my experience it is easier for humans to parse code lines that are as short and clear as possible. Multiple conditionals within the same line are harder to parse for some people.

Reply:
ok for me it's the opposite but ok

Reply:
I personally find this function pretty clean. Depends on a human apparently but I agree with @xh3b4sd here.